hadoop-ozone/dev-support/checks/sonar.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+cd "$DIR/../../.." || exit 1
+
+if [ ! "$SONAR_TOKEN" ]; then
+  echo "SONAR_TOKEN environment variable should be set"
```

Comment:
Can we echo in what format this should be --? export SONAR_TOKEN=<token> ? or do wee need anything else.
and also a line about you can get your sonar token from the sonar cloud.io if you are added as part of the Ozone group. I will commit this now, but just a suggestion so that users know how to get the token and how to export it. We also add that info in the Wiki and print a pointer to it. Or we can add a README.txt in the tools directory. Just wanted a way for others who are not familiar with this process to find a way forward. Thanks

Reply:
Ideally, we don't want users to run sonar.sh in their local environments with SONAR_TOKEN, since the results will be published to sonar cloud. Now since the CI will take care of publishing results to sonarcloud after every commit, I would strongly discourage anyone to run sonar in dev setups. Devs can use this plugin to check sonar issues during development - https://plugins.jetbrains.com/plugin/7973-sonarlint. We should add this to our README and Contributing guide.

Reply:
Ok..makes sense. We can add than info to this tool later.